packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -349,24 +344,25 @@ private function isObjectTypeOfObjectType(ObjectType $resolvedObjectType, Object
             return true;
         }
 
-        if (! $this->reflectionProvider->hasClass($resolvedObjectType->getClassName())) {
+        if (! $this->reflectionProvider->hasClass($requiredObjectType->getClassName())) {
             return false;
         }
+        $requiredClassReflection = $this->reflectionProvider->getClass($requiredObjectType->getClassName());
 
-        $classReflection = $this->reflectionProvider->getClass($resolvedObjectType->getClassName());
-        if (! isset($this->traitExistsCache[$classReflection->getName()])) {
-            $this->traitExistsCache[$classReflection->getName()] = \trait_exists($requiredObjectType->getClassName());
+        if (! $this->reflectionProvider->hasClass($resolvedObjectType->getClassName())) {
+            return false;
         }
+        $resolvedClassReflection = $this->reflectionProvider->getClass($resolvedObjectType->getClassName());
 
-        if ($this->traitExistsCache[$classReflection->getName()]) {
-            foreach ($classReflection->getAncestors() as $ancestorClassReflection) {
```

Comment:
this ancestor traversal is slow. after this PR we only do it for trait-classes. before we did it for every class.